### PR TITLE
[READY] Metal compactor, works for more than just steel, cannot transmute materials

### LINKED ISSUE #24843: Recycler only recycles steel
The Steel Compactor (t_recycler) only recycles steel items into steel materials (steel_lump, sheet_metal, steel_chunk, & scrap) currently.

Now with the addition of the recycle center in-game it would be a nice feature for the Steel Compactor to be an all around recycler... recycling other metal (copper, iron, lead, & aluminum) items and maybe plastic and glass items into their material components.

Code for the current recycling process can be found on line 2713 in iexamine.cpp

### LINKED ISSUE #25013: Steel compactor able to transmute other materials to steel
The steel compactor `t_recycler` is described as a piece of machinery that 
> accept[s] inputs of various metals and through a complex system of hydraulics, [reshape them] into more useful parts and pieces.

(Currently it only recycles steel, but I am working on a PR for #24843.)

**Expected behavior**  
- It should not be able to automagically isolate only items made from steel from a large pile of items fed to it as input
- It has no means of deconstructing items into component parts -- crushing the whole thing doesn't count -- and thus should not accept items that are made from multiple materials
- If such a machine IRL were fed with items of mixed materials, we would expect the product to be contaminated, e.g. splinters of wood crushed and embedded in output metal ingots and such.
- It should certainly not have transmutative powers of turning one material into another.

**Actual behavior**  
- Is able to pick out only items that have steel content from mixed input
- Will accept items of mixed material (e.g. steel+wood)
  - still considers the weight of entire item for calculations
  - will effectively turn wood into steel

**To Reproduce**  
1. debug and map edit yourself a steel compactor
2. wish for the `homewrecker` item (mixed Steel + Wood)
3. feed it to the compactor. estimate is 9 chunks of steel, can produce more in practice

Consider that the homewrecker is craftable at fabrication (0) from:
- 8 long string / 4 small string / 96 withered plant / 96 piles of straw
- chunk of steel (4)
- 1 heavy stick / 1 two by four / 1 broom

This becomes an avenue that can be exploited to convert a renewable resource (wood) into one which can only be scavenged (steel).

### PR DESCRIPTION
**Pull request by cake-pie**
Resolve #24843, resolve #25013

Still WIP, but it kind of works already, so putting it out for folks to try/test.
I've left the original code (`iexamine::recycler()`) in place for now and written new code separately (`iexamine::recycle_compactor`) so it's easy to switch between the two behaviors by editing `terrain.json` and changing `t_recycler`:

> ~"examine_action": "recycler",~
> "examine_action": "recycle_compactor",

For this to have completely sane behavior will require fixing quirky stats on various metal objects.

Enforcing the input to be pure metal is more inconvenient, but I plan to alleviate that with additional recycling mechanics (shredder to tear items apart; electromagnet to isolate items with ferrous content).

### CODE REVIEW

File: src/iexamine.cpp [hunk 1]
```diff
@@ -2674,6 +2675,104 @@ void iexamine::shrub_wildveggies( player &p, const tripoint &examp )
...
+        return;
+    }
+    material_type m = metals.at(m_idx);
+
+    // check inputs and tally total mass
+    auto inputs = g->m.i_at(examp);
+    units::mass sum_weight = 0;
+    for( auto it = inputs.begin(); it != inputs.end(); ++it ) {
+        if( !it->only_made_of( std::set<material_id> { m.id } ) ) {
+            add_msg(_("You realize this isn't going to work because %s is not made purely of %s."), it->tname().c_str(), m.name().c_str());
```
Comment (Night-Pryanik) [inline src/iexamine.cpp]: Please use positional arguments here and below. Consider also adding translator comments.

**Comment by cake-pie** [comment]
I had a sneaking suspicion that the math I copied from the original wasn't quite doing the correct thing.
Turns out the issue was with output items that had default charges > 1. Fixed.

Now accepts hard steel, budget steel, and iron as inputs for steel output, as they don't have suitable outputs of their own, and per https://github.com/CleverRaven/Cataclysm-DDA/issues/24843#issuecomment-414644513

**Comment by cake-pie** [comment]
All set to go.

File: data/json/materials.json [hunk 2]
```diff
@@ -746,7 +751,9 @@
     "dmg_adj": [ "marked", "dented", "smashed", "shattered" ],
     "bash_dmg_verb": "dented",
     "cut_dmg_verb": "scratched",
-    "burn_products": [ [ "scrap", 1 ] ]
+    "burn_products": [ [ "scrap", 1 ] ],
+    "compact_accepts": [ "budget_steel", "hardsteel", "iron" ],
+    "compacts_into": [ "steel_lump", "sheet_metal", "steel_chunk", "scrap" ]
```
Comment (ZhilkinSerg) [inline data/json/materials.json]: I am not sure `sheet_metal` is proper output.

Comment (cake-pie) [reply data/json/materials.json]: This preserves status quo: the original steel compactor also provided these four output options.

Comment (cake-pie) [reply data/json/materials.json]: Willing to remove `sheet_metal` (and by extension, `aluminum_foil`) if you insist -- let me know.

**Pull request merged**